scene/resources/surface_tool.cpp

```diff
@@ -85,13 +85,21 @@ uint32_t SurfaceTool::VertexHasher::hash(const Vertex &p_vtx) {
 	return h;
 }
 
-void SurfaceTool::begin(Mesh::PrimitiveType p_primitive) {
+void SurfaceTool::begin(Mesh::PrimitiveType p_primitive, VisualServer::ArrayFormat p_format_flags) {
```

Comment:
Just use a boolean in SurfafceTool
set_use_8_bone_weights(bool) and of course throw an error if the surface is not empty when setting it.